common/djangoapps/lang_pref/middleware.py

```diff
@@ -26,3 +28,12 @@ def process_request(self, request):
             # Set it to the LANGUAGE_SESSION_KEY (Django-specific session setting governing language pref)
             if user_pref:
                 request.session[LANGUAGE_SESSION_KEY] = user_pref
+            else:
+                # Set browser language if it is supported by system.
```

Comment:
We are not actually setting the browser language here -- we are setting the session language to the browser language, if it is supported.